distsql/distsql.go

```diff
@@ -183,17 +188,16 @@ func (r *selectResult) getSelectResp() (*tipb.SelectResponse, error) {
 	}
 }
 
-func (r *selectResult) readRowsData(chk *chunk.Chunk, rowsData []byte) error {
-	var err error
-	for len(rowsData) > 0 {
+func (r *selectResult) readRowsData(chk *chunk.Chunk, rowsData []byte) (remain []byte, err error) {
+	for chk.NumRows() < r.ctx.GetSessionVars().MaxChunkSize && len(rowsData) > 0 {
 		for i := 0; i < r.rowLen; i++ {
-			rowsData, err = codec.DecodeOneToChunk(rowsData, chk, i, r.fieldTypes[i], r.loc)
+			rowsData, err = codec.DecodeOneToChunk(rowsData, chk, i, r.fieldTypes[i], r.ctx.GetSessionVars().GetTimeZone())
```

Comment:
How about assigning TimeZone and MaxChunkSize to local variables outside the for loop?